### PR TITLE
feat: extract domain from url

### PR DESCRIPTION
**Pull request by rem1niscence**

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"net"
+	"net/url"
+	"strings"
+)
+
+var (
+	errInvalidURL = errors.New("invalid URL")
+)
+
+func extractDomain(inputURL string) (string, error) {
```
Comment (pablocampogo) [inline domain/domain.go:14]: Good luck using that private function from somewhere else 😁

File: domain/domain.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	if len(parts) > 2 && !isIP(parts[len(parts)-2]) {
+		// If the last two parts of the hostname are not an IP address, join the last two parts
+		return strings.Join(parts[len(parts)-2:], "."), nil
+	}
+	// Otherwise, return the full hostname
+	return parsedURL.Hostname(), nil
+}
+
+// isIP checks whether the given string is a valid IP address
+func isIP(str string) bool {
```
Comment (pablocampogo) [inline domain/domain.go:33]: Make this public too

**Review by pablocampogo — CHANGES_REQUESTED** [review]

File: domain/domain.go (hunk 1 above)
Comment (rem1niscence) [reply domain/domain.go]: 🤓

File: domain/domain.go (hunk 2 above)
Comment (rem1niscence) [reply domain/domain.go:37]: that one is not intended to be public

**Review by pablocampogo — APPROVED** [review]

**Pull request merged**